data/org.eclipse.birt.report.data.oda.jdbc/src/org/eclipse/birt/report/data/oda/jdbc/JDBCDriverManager.java

```diff
@@ -313,6 +317,29 @@ private Connection doConnect(String driverClass, String url, String jndiNameUrl,
 		}
 	}
 
+	/*
+	 * Remove unsupported properties from the connection properties
+     * based on the driver property information set
+	 */
+	private void removeUnsupportedConnectionProperties(Driver driver, String url, Properties connectionProperties) {
+		try {
+			DriverPropertyInfo[] supportedProperties = driver.getPropertyInfo(url, connectionProperties);
+			if (supportedProperties != null && connectionProperties != null) {
+				connectionProperties.forEach((key, value) -> {
+					boolean isPropertySupported = false;
+					for (DriverPropertyInfo supportedProperty : supportedProperties) {
+						if (supportedProperty.name.equalsIgnoreCase(key.toString()))
+							isPropertySupported = true;
```

Comment:
No big deal, but you could just return at this point, and eliminate the isPropertySupported boolean variable along with the guard after the loop.  Also one could of course break the loop once this comes true.

Reply:
Good hint to break the loop after the match.
I'm not sure with your idea of the boolean-variable because it is part of the inner loop - what is your idea with "guard after the loop".

Reply:
I can make a better suggestion but I'm in the middle testing derby and cannot do two things at once.